llvm/test/Transforms/LoopVectorize/debugloc-optimize-vfuf-term.ll

```diff
@@ -0,0 +1,108 @@
+; CHECK-NEXT:    br i1 [[TMP4]], label %[[PRED_STORE_IF3:.*]], label %[[PRED_STORE_CONTINUE4]], !dbg [[DBG9]]
+; CHECK:       [[PRED_STORE_IF3]]:
+; CHECK-NEXT:    [[TMP5:%.*]] = trunc i64 0 to i8, !dbg [[DBG8]]
+; CHECK-NEXT:    store i8 [[TMP5]], ptr null, align 1, !dbg [[DBG9]]
+; CHECK-NEXT:    br label %[[PRED_STORE_CONTINUE4]], !dbg [[DBG9]]
+; CHECK:       [[PRED_STORE_CONTINUE4]]:
+; CHECK-NEXT:    [[INDEX_NEXT]] = add i64 [[INDEX]], 2, !dbg [[DBG4]]
+; CHECK-NEXT:    br i1 true, label %[[MIDDLE_BLOCK:.*]], label %[[VECTOR_BODY]], !dbg [[DBG4]], !llvm.loop [[LOOP10:![0-9]+]]
+; CHECK:       [[MIDDLE_BLOCK]]:
+; CHECK-NEXT:    br i1 true, label %[[DO_BODY45_PREHEADER:.*]], label %[[SCALAR_PH]], !dbg [[DBG13:![0-9]+]]
+; CHECK:       [[SCALAR_PH]]:
+; CHECK-NEXT:    [[BC_RESUME_VAL:%.*]] = phi i64 [ [[N_VEC]], %[[MIDDLE_BLOCK]] ], [ 0, %[[ENTRY]] ], !dbg [[DBG4]]
+; CHECK-NEXT:    br label %[[DO_BODY:.*]]
+; CHECK:       [[DO_BODY]]:
+; CHECK-NEXT:    [[INDVARS_IV554:%.*]] = phi i64 [ [[INDVARS_IV_NEXT555:%.*]], %[[DO_BODY]] ], [ [[BC_RESUME_VAL]], %[[SCALAR_PH]] ], !dbg [[DBG4]]
+; CHECK-NEXT:    [[CONV39:%.*]] = trunc i64 0 to i8, !dbg [[DBG8]]
+; CHECK-NEXT:    store i8 [[CONV39]], ptr null, align 1, !dbg [[DBG9]]
+; CHECK-NEXT:    [[INDVARS_IV_NEXT555]] = add i64 [[INDVARS_IV554]], 1, !dbg [[DBG14:![0-9]+]]
+; CHECK-NEXT:    [[EXITCOND557_NOT:%.*]] = icmp eq i64 [[INDVARS_IV554]], [[WIDE_TRIP_COUNT]], !dbg [[DBG15:![0-9]+]]
+; CHECK-NEXT:    br i1 [[EXITCOND557_NOT]], label %[[DO_BODY45_PREHEADER]], label %[[DO_BODY]], !dbg [[DBG13]], !llvm.loop [[LOOP16:![0-9]+]]
+; CHECK:       [[DO_BODY45_PREHEADER]]:
+; CHECK-NEXT:    ret i32 0
+;
+entry:
+  %wide.trip.count = and i64 %a, 1
+  br label %do.body
+
+do.body:                                          ; preds = %do.body, %entry
+  %indvars.iv554 = phi i64 [ %indvars.iv.next555, %do.body ], [ 0, %entry ], !dbg !4
+  %conv39 = trunc i64 0 to i8, !dbg !5
+  store i8 %conv39, ptr null, align 1, !dbg !6
+  %indvars.iv.next555 = add i64 %indvars.iv554, 1, !dbg !7
+  %exitcond557.not = icmp eq i64 %indvars.iv554, %wide.trip.count, !dbg !8
+  br i1 %exitcond557.not, label %do.body45.preheader, label %do.body, !dbg !9
+
+do.body45.preheader:                              ; preds = %do.body
+  ret i32 0
+}
+
+!llvm.dbg.cu = !{!0}
+!llvm.module.flags = !{!3}
+
+!0 = distinct !DICompileUnit(language: DW_LANG_C_plus_plus_14, file: !1, producer: "clang version 20.0.0git (https://github.com/llvm/llvm-project.git 97cea5fecc5fa70842644da877c9547d4f34f6db)", isOptimized: true, runtimeVersion: 0, emissionKind: FullDebug, enums: !2, retainedTypes: !2, globals: !2, imports: !2, splitDebugInlining: false, nameTableKind: None)
+!1 = !DIFile(filename: "/home/gbtozers/dev/llvm-test-suite/CTMark/7zip/CPP/7zip/Compress/BZip2Decoder.cpp", directory: "/home/gbtozers/dev/llvm-test-suite-build", checksumkind: CSK_MD5, checksum: "ace37523c35f8079a0f00d337fc014dc")
```

Comment:
Not sure if any of this can be shortened.

Reply:
Yes it can - most of the tests I cleaned the metadata up before submitting, unfortunately look like I missed this one!